app/src/main/java/me/guillaumin/android/osmtracker/gpx/ExportAndShareTask.java

```diff
@@ -0,0 +1,168 @@
+package me.guillaumin.android.osmtracker.gpx;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.database.Cursor;
+import android.net.Uri;
+import android.util.Log;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Date;
+import java.util.TreeMap;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import me.guillaumin.android.osmtracker.R;
+import me.guillaumin.android.osmtracker.exception.ExportTrackException;
+
+/**
+ * Exports the track and media (if any) to a a temporary file, then shares it with another app
+ */
+
+public class ExportAndShareTask extends ExportTrackTask {
+    private static final String TAG = ExportToTempFileTask.class.getSimpleName();
+
+    /**
+     * Maps from name that is passed in URI to another app to a real file in the cache dir.
+     */
+    static final TreeMap<String, File> sharedFiles = new TreeMap<String, File>();
+    /**
+     * Name of the file, passed to another app in URI.
+     */
+    private String filename;
+    /**
+     * Subfolder in cacheDir, where track is exported
+     */
+    private File trackDir;
+
+    public ExportAndShareTask(Context context, long trackId) {
+        super(context, trackId);
+        File cacheDir = context.getCacheDir();
+        // Generate name for a temporary directory
+        trackDir = new File(cacheDir, "trackDirectory");
```

Comment:
Use a constant for `trackDirectory`?

Also, should we take care of cleaning up after ourselves? I'm concerned the cache folder will grow up over time.